reactive/calico.py

```diff
@@ -323,22 +332,38 @@ def ready():
 
 @when('config.changed.registry-credentials')
 def registry_credentials_changed():
-    status_set('maintenance', 'Applying registry credentials')
-    encoded_creds = hookenv.config('registry-credentials')
+    remove_state('calioco.image.pulled')
```

Comment:
We'll also need a handler that reacts to `config.changed.registry` by clearing `calioco.image.pulled`

Reply:
We also should clear `calioco.image.pulled` during the upgrade_charm handler.